### DISCUSSION TITLE
gum choose in a fancy box

### DISCUSSION THREAD

**Discussion by dragonis41** [charmbracelet/gum#417] [Q&A]
Hi there,
I have read every example and community code but I still can't figure out if it's possible to concatenate commands.
For example: display the item picker (gum choose) in a fancy box centered in my terminal (gum style).

Is there a way to do that, or is it not yet implemented ?

**Comment by wellkilo** [comment]
gum choose and gum style cannot be composed as a simple pipe for this particular layout.

The reason is that gum choose is interactive and owns the terminal while it is running; its stdout is only the **selected value after exit**. Therefore this:

    gum choose ... | gum style ...

does not give gum style the live chooser UI—it only styles the final selection.

For the parts that choose itself exposes, style it directly with its flags and environment variables, for example:

    choice=$(gum choose \
      --header 'Choose an item' \
      --height 10 \
      --padding '1 4' \
      --cursor.foreground 212 \
      --selected.foreground 212 \
      one two three)

    printf 'Selected: %s\n' "$choice"

Current choose supports padding plus header/cursor/item/selected styles, but it has no border, width, or whole-widget alignment option. Those are options of gum style, which renders static text read from arguments/stdin. You can see the actual choose options in [choose/options.go](https://github.com/charmbracelet/gum/blob/7179388031ae67d7f538d001be87d931f1cf5e28/choose/options.go) and style's static-input behavior in [style/command.go](https://github.com/charmbracelet/gum/blob/7179388031ae67d7f538d001be87d931f1cf5e28/style/command.go).

So a live bordered, terminal-centered chooser is not supported by concatenating the two CLI commands. For that exact UI, either use choose's built-in styling as far as it goes, or build a small Bubble Tea/Lip Gloss program where the list and border are part of one model.
